server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -629,6 +635,30 @@ private void cleanupOfflineMigrations() {
         }
       }
     }
+
+    /**
+     * Remove any migrations to any of the deleted TServers
+     */
+    private void cleanupDeletedMigrations() {
+      synchronized (deletedTServers) {
+        var iter = deletedTServers.iterator();
+        var ample = getContext().getAmple();
+        while (iter.hasNext()) {
+          var deletedTServer = iter.next();
+          for (DataLevel dl : DataLevel.values()) {
+            // prev row needed for the extent
+            try (var tabletsMetadata = ample.readTablets().forLevel(dl)
+                .fetch(TabletMetadata.ColumnType.PREV_ROW, TabletMetadata.ColumnType.MIGRATION)
+                .build()) {
+              for (var tabletMetadata : tabletsMetadata) {
+                conditionallyDeleteMigration(tabletMetadata.getExtent(), deletedTServer);
+              }
+            }
+          }
+          iter.remove();
+        }
+      }
+    }
```

Comment:
Tracking these deleted tservers is probably not needed, can use the set of online tablet servers instead.  Also would be better to create a single tablet mutator and use it for all tablets instead of creating one per tablet.  This allows streaming data from the metadata table to tablet server write ahead logs.

```suggestion
        var ample = getContext().getAmple();
          for (DataLevel dl : DataLevel.values()) {
            // prev row needed for the extent
            try (var tabletsMetadata = ample.readTablets().forLevel(dl)
                .fetch(TabletMetadata.ColumnType.PREV_ROW, TabletMetadata.ColumnType.MIGRATION)
                .build(); var tabletsMutator = ample.conditionallyMutateTablets(result->{})) {
              for (var tabletMetadata : tabletsMetadata) {
                if(tabletMetadata.getMigration() != null && !onlineTabletServers().contains(tabletMetadata.getMigration())) {
                 tabletsMutator.mutateTablet(tabletMetadata.getExtent()).requireAbsentOperation().requireMigration(tabletMetadata.getMigration()).deleteMigration().submit(tm->false);
                }
              }
            }
          }
```